[12] - Implement the authorization/authentication functionality

spatiszisis opened the pull request:
- Added some new libraries that needed for this issue. (e.g. spring-boot-starter-security etc.)
- Introduced some new routes for the new auth operations
- Introduced a whole new functionality for the Authorization/Authentication process of the project
- Introduced a new entity to store the tokens of the users.

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Added some new libraries that needed for this issue. (e.g. spring-boot-starter-security etc.)
- Introduced some new routes for the new auth operations
- Introduced a whole new functionality for the Authorization/Authentication process of the project
- Introduced a new entity to store the tokens of the users.

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Made some structure modifications on the project

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Some extra minor modification

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Some extra minor modifications

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Some extra minor modifications
- Deleted an unused file

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- merge branch with latest changes from dev branch

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Added some missing staff from the entities, converters and services
- Added some annotations to prevent to write extra code. (e.g. for constructors)
- Fixed some minor stylistic issues.

EmmanSkout reviewed (approved):
In general, very well implemented and clean code. 

I like the use you are making of annotations and more specifically Lombok . 
Maybe we could cut down the use of "var" . In my opinion , java is meant to be strongly typed and we should stick to that. 

Other than that, I'm happy with this request and I'm approving.

apokiou reviewed (approved):
Looks Good Zisis!

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Replaced all the vars with the corresponding types of each value.

christosavramis commented on src/main/java/net/trapezokomos/dashboard/controller/ConsumerController.java:
Change request param annotation to path variable when applicable e.g
 @RequestParam(value = "id", required = true) Long id 
to @PathVariable Long id



request param is ?param
path variable param is /param/
also in general avoid setting default values e.g. required = true

spatiszisis replied:
OK

spatiszisis committed:
#12 - Implement the Authorization/Authentication functionality
- Added an extra rule for the .gitignore file
- Added some extra modifications e.g remove/change some annotations
- Added a script to run the docker commands
- Modified the README.md file with some extra changes
